spec/models/assignment_spec.rb

```diff
@@ -2227,6 +2227,42 @@ def grouping_count(groupings)
     end
   end
 
+  describe '#get_num_valid' do
+    before :each do
+      @assignment = create(:assignment, assignment_properties_attributes: { group_min: 2, group_max: 3 })
```

Comment:
So for new tests, we prefer using `let`/`let!` instead of instance variables. You can change this to something like `let(:assignment) { create(:assignment, ...) }`, and then the variable `assignment` will be available for use in the tests.

You also do *not* need the `before :each` block when using `let`/`let!`.